src/redux/reducers/messages.js

```diff
@@ -0,0 +1,125 @@
+import { handleActions } from 'redux-actions';
+import actionTypes from '../actionTypes';
+
+const defaultState = {
+  messages: [],
+  messagesByID: {},
+  unreadMessagesByID: {},
```

Comment:
`unreadMessageIds: []`

Change the logic accordingly.

Reply:
@akshatnitd We won't be able to search for a message by its id in O(1) time if we convert it to an array. Waiting for further clarification from your side.

Reply:
eg - 
```
messagesById : {
    1 : {
            "text": "Hi"
    },
    2 : {
            "text": "Hi"
    },
},
unreadMessageIds : [ 2 ],
}
``` 

Therefore info about unreadMessage - 
`messagesById[unreadMessageIds[0]]`

Reply:
It can be done in o(1)